backend/tests/test_amd_gpu_detect.py

```
@@ -0,0 +1,89 @@
+    )
+    @patch("torch.cuda.is_available", return_value=False)
+    def test_no_torch_cuda(self, _mock_avail, _mock_run, _mock_system):
+        """When WMI fails and torch.cuda is unavailable, should return False."""
+        assert is_amd_gpu_windows() is False
+
+    @patch("utils.platform_detect.platform.system", return_value="Windows")
+    @patch(
+        "utils.platform_detect.subprocess.run",
+        side_effect=Exception("WMI not available"),
+    )
+    def test_torch_not_installed(self, _mock_run, _mock_system):
```

Ruff (0.15.10)

[warning] Fixture `_mock_run` without value is injected as parameter, use `@pytest.mark.usefixtures` instead

(PT019)